src/DynamoDb/Marshaler.php

```diff
@@ -0,0 +1,294 @@
+<?php
+namespace Aws\DynamoDb;
+
+use GuzzleHttp\Stream\StreamInterface;
+
+/**
+ * Marshals and unmarshals JSON documents and PHP arrays into DynamoDB items.
+ */
+class Marshaler
+{
+    /** @var array Default options to merge into provided options. */
+    private static $defaultOptions = [
+        'ignore_invalid'  => false,
+        'wrap_numbers'    => false,
+        'invalid_handler' => null,
+    ];
+
+    /** @var array Marshaler options. */
+    private $options;
+
+    /**
+     * Instantiates a DynamoDB Marshaler.
+     *
+     * The following options are valid.
+     *
+     * - skip_invalid: (bool) Set to `true` if invalid values should be skipped
+     *   (i.e., not included) during marshaling.
+     * - wrap_numbers: (bool) Set to `true` to wrap numbers with `NumberValue`
+     *   objects during unmarshaling to preserve the precision of numbers.
+     * - invalid_handler: (callable) Allows user to provide custom logic to
```

Comment:
Well, they are 2 solutions to the same problem. With ignore_invalid, they are simply dropped from the marshaled data. With invalid_handler, you can massage the data (e.g., convert empty strings to null values). I'm not sure I would want to remove one or the other. What do you think?